x-pack/plugins/alerting_builtins/server/alert_types/geo_threshold/es_query_builder.ts

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ILegacyScopedClusterClient } from 'kibana/server';
+
+export const OTHER_CATEGORY = 'other';
+
+async function getShapesFilters(
+  boundaryIndexTitle: string,
+  boundaryType: string,
+  boundaryGeoField: string,
+  geoField: string,
+  callCluster: ILegacyScopedClusterClient['callAsCurrentUser'],
+  log: Logger
+) {
+  const filters = {};
+  switch (boundaryType) {
+    case 'entireIndex':
+      // Get all shapes in index
+      const boundaryData = await callCluster('search', {
+        index: boundaryIndexTitle,
+        body: {},
+      });
```

Comment:
I believe by default you will only get 10 entries.  And if you want to get more, curious how we would limit the number we search for.  It could be another parameter on the alert, or could be a config property, or ???

Reply:
Nice catch! For this PR I've set a default to 10K for now with a comment that we may want to make this value configurable in the future, maybe similar to how we've done with visualizations. Not sure for now if that value would be specific to geo-alerts, all built-ins or something else entirely but definitely worth circling back around on